src/Server/Sanitizer.fs

```diff
@@ -13,59 +13,186 @@ let unsafe = ["System"; "NativePtr"]
 let isLongIdentIncludeUnsafe (ident: LongIdent) =
   ident |> Seq.exists (fun i -> List.contains i.idText unsafe)
 
+let isTypeInheritUnsafe = function
+  | SynType.LongIdent (LongIdentWithDots(ident, _)) ->
+    ident |> isLongIdentIncludeUnsafe
+  | SynType.LongIdentApp (_, LongIdentWithDots(ident, _), _, _, _, _, _) ->
+    ident |> isLongIdentIncludeUnsafe
+  | _ -> false
+
 let rec visitExpr = function
+  // Application
   | SynExpr.App (_, _, f, arg, _) ->
     visitExpr f
     || visitExpr arg
+
+  // Long Identifier
   | SynExpr.LongIdent (_, LongIdentWithDots (ident, _), _, _) ->
     ident |> isLongIdentIncludeUnsafe
-  | SynExpr.Match (_, e, clauses, _) ->
+
+  // Match expressions
+  | SynExpr.Match (_, e, clauses, _)
+  | SynExpr.MatchBang (_, e, clauses, _) ->
     visitExpr e
-    || clauses |> List.fold (fun acc (Clause (_, wh, e, _, _)) ->
-      acc
-      || visitExpr e
-      || (match wh with | Some e -> visitExpr e | None -> false)) false
+    || clauses |> List.fold visitClause false
+  | SynExpr.MatchLambda (_, _, clauses, _, _) ->
+    clauses |> List.fold visitClause false // Maybe List.exists?
+
+  // If-Then-Else expression
   | SynExpr.IfThenElse (g, t, f, _, _, _, _) ->
     visitExpr g
     || visitExpr t
     || (match f with | Some f -> visitExpr f | None -> false)
+
+  // Let/Use expression
   | SynExpr.LetOrUse (_, _, bindings, body, _) ->
     bindings
-    |> List.forall (fun (Binding (_, _, _, _, _, _, _, _, _, e, _, _)) ->
+    |> List.forall (fun (Binding (_, _, _, _, _, _, _, _, _, e, _, _)) -> // Why is this forall?
```

Comment:
For all expression, we want to have true. If there is at least one false, the whole evaluation should return false.

Reply:
Same as above comment, PoC as an example:
```f#
let rec f1 =
  System.Console.WriteLine "System"
and f2 =
  ()
```

Reply:
Yes it was a bug. Need to fix this.